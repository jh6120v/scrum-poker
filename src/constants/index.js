export const POINT_SEQUENCE_LIST = 'POINT_SEQUENCE_LIST';
export const PERSONAL_SETTING = 'PERSONAL_SETTING';
export const FUNC_GO_BACK = 'GO_BACK';
export const FUNC_CLOSE = 'CLOSE';
export const FUNC_SETTING = 'SETTING';
export const PERSONAL_DEFAULT_SETTING = {
    sequenceType: '',
    theme: {
        cardList: {
            borderColor: '#FFFFFF',
            bgColor: '#5bb1d4',
            fontColor: '#FFFFFF'
        },
        cardFront: {
            bgColor: '#FFFFFF',
            fontColor: '#5AC9E8'
        },
        cardBack: {
            borderColor: '#FFFFFF',
            bgColor: '#5bb1d4',
            fontColor: '#FFFFFF'
        }
    },
    keepScreenOn: false,
    autoHideCard: false
};
