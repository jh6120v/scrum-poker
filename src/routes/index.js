import React from 'react';
import { Route, Switch, withRouter } from 'react-router';
import { CSSTransition, TransitionGroup } from 'react-transition-group';
import { RouterConfig, getSceneConfig } from './router-config';
import { RouterAnimation, RouteAnimateContent } from '../styles/router-animation';

let oldLocation = null;
const Routes = withRouter(({ location, history }) => {
    let classNames = '';
    if (history.action === 'PUSH') {
        classNames = `page ${getSceneConfig(location).enter}`;
    } else if (history.action === 'POP' && oldLocation) {
        classNames = `page ${getSceneConfig(oldLocation).exit}`;
    }

    oldLocation = location;

    return (
        <RouterAnimation>
            <TransitionGroup
                childFactory={(child) => React.cloneElement(child, { classNames })}
            >
                <CSSTransition
                    timeout={300}
                    key={location.pathname}
                >
                    <RouteAnimateContent>
                        <Switch location={location}>
                            {
                                RouterConfig.map((config) => (
                                    <Route
                                        key={config.path}
                                        path={config.path}
                                        component={config.component}
                                        exact={config.exact}
                                    />
                                ))
                            }
                        </Switch>
                    </RouteAnimateContent>
                </CSSTransition>
            </TransitionGroup>
        </RouterAnimation>
    );
});

export default Routes;
