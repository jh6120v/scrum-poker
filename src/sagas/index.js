import { takeEvery, all } from 'redux-saga/effects';
import { indexDBDel, indexDBSet } from './indexed-db';
import { IDBDelete, IDBSet } from '../modules/indexed-db';

function* watchIndexDBSet() {
    yield takeEvery(IDBSet.type, indexDBSet);
}

function* watchIndexDBDel() {
    yield takeEvery(IDBDelete.type, indexDBDel);
}

export default function* rootSaga() {
    yield all([
        watchIndexDBSet(),
        watchIndexDBDel()
    ]);
}
