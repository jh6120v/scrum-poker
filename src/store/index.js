import { createStore, applyMiddleware, compose } from 'redux';
import { routerMiddleware } from 'connected-react-router';
import { createBrowserHistory } from 'history';
import createSagaMiddleware from 'redux-saga';
import createRootReducer from './reducers';

export const history = createBrowserHistory();
export const sagaMiddleware = createSagaMiddleware();

export function ConfigureStore(initialState = {}) {
    // ======================================================
    // Store Enhancers
    // ======================================================
    let composeEnhancer;
    console.log(process.env.NODE_ENV);
    if (process.env.NODE_ENV === 'development') {
        composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
    } else {
        composeEnhancer = compose;
    }

    const store = createStore(
        createRootReducer(history, {}), // root reducer with router state
        initialState,
        composeEnhancer(
            applyMiddleware(
                routerMiddleware(history),
                sagaMiddleware
            )
        )
    );

    store.asyncReducers = {};

    return store;
}

export const store = ConfigureStore();

if (module.hot) {
    module.hot.accept('./reducers', () => {
        const reducers = require('./reducers').default;
        store.replaceReducer(reducers(store.asyncReducers));
    });
}
