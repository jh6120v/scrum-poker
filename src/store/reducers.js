import { combineReducers } from 'redux';
import { connectRouter } from 'connected-react-router';
import { identity, prop, propOr } from 'ramda';

const createRootReducer = (history, asyncReducers) => combineReducers({
    router: connectRouter(history),
    ...asyncReducers
});

export const injectReducer = (history, store, reducerList = []) => {
    // console.log(reducerList);
    const injectList = store.asyncReducers;
    for (const item of reducerList.values()) {
        injectList[item.key] = item.reducer;
    }

    store.replaceReducer(createRootReducer(history, injectList));
};

export const createReducer = (initialState, handlers) => (
    state = initialState,
    action
) => propOr(identity, prop('type', action), handlers)(state, action);

export default createRootReducer;
