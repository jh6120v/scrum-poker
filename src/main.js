import '@babel/polyfill';
import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import ResetStyle from './styles/reset-style';
import GlobalStyle from './styles/global-style';
import App from './containers/app';
import { store } from './store';

if ('serviceWorker' in navigator) {
    window.addEventListener('load', () => {
        navigator.serviceWorker.register('/service-worker.js').then((registration) => {
        }).catch((registrationError) => {
            console.log('SW registration failed: ', registrationError);
        });
    });
}

ReactDOM.render(
    <Provider store={store}>
        <ResetStyle />
        <GlobalStyle />
        <App />
    </Provider>,
    document.getElementById('app')
);

if (module.hot && process.env.NODE_ENV !== 'production') {
    module.hot.accept();
}
