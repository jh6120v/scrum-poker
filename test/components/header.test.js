import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import { Router } from 'react-router';
import Header from '../../src/components/header';
import { FUNC_CLOSE, FUNC_GO_BACK, FUNC_SETTING } from '../../src/constants';
import { history } from '../../src/store';

describe('test header component.', () => {
    it('test <Header /> snapshot.', () => {
        cleanup();

        const { container } = render(<Header />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <Header /> when set title.', () => {
        cleanup();

        const { getByTestId } = render(<Header title="test_title" />);

        expect(getByTestId('display_title').innerHTML).toEqual('test_title');
    });

    it('test <Header /> when go back btn exist.', () => {
        cleanup();

        const { getByTestId } = render(<Header prev={FUNC_GO_BACK} />);

        expect(getByTestId('display_prev_link').querySelector('a').innerHTML).toEqual('<svg class="sc-bdVaJa fUuvxv" fill="#fff" width="25px" height="25px" viewBox="0 0 1024 1024" rotate="0"><path d="M704 256.8l-64.6-64.8-319.4 320 319.4 320 64.6-64.8-254.6-255.2z"></path></svg>');
    });

    it('test <Header /> when go back click once', () => {
        cleanup();

        history.goBack = jest.fn();

        const { getByTestId } = render(<Header prev={FUNC_GO_BACK} />);

        fireEvent.click(getByTestId('display_prev_link').querySelector('a'));

        expect(history.goBack).toHaveBeenCalledTimes(1);
    });

    it('test <Header /> when close btn exist.', () => {
        cleanup();

        const { getByTestId } = render(<Header prev={FUNC_CLOSE} />);

        expect(getByTestId('display_prev_link').querySelector('div').innerHTML).toEqual('<svg class="sc-bdVaJa fUuvxv" fill="#fff" width="25px" height="25px" viewBox="0 0 1024 1024" rotate="0"><path d="M810 273.596l-59.596-59.596-238.404 238.404-238.404-238.404-59.596 59.596 238.404 238.404-238.404 238.404 59.596 59.596 238.404-238.404 238.404 238.404 59.596-59.596-238.404-238.404z"></path></svg>');
    });

    it('test <Header /> when setting btn exist.', () => {
        cleanup();

        const { getByTestId } = render(<Router history={history}><Header next={FUNC_SETTING} /></Router>);

        expect(getByTestId('display_next_link').querySelector('div').innerHTML).toEqual('<a href="/setting"><svg class="sc-bdVaJa fUuvxv" fill="#fff" width="25px" height="25px" viewBox="0 0 1024 1024" rotate="0"><path d="M822.2 512c0-47.8 29.6-85.6 73.8-111.6-8-26.6-18.6-52.4-31.6-76.4-49.8 13-90-6.4-124-40.4-33.8-33.8-44.2-74.2-31.2-124-24-13-49.6-23.6-76.4-31.6-26 44.4-72.8 73.8-120.8 73.8-47.8 0-94.8-29.4-120.8-73.8-26.8 8-52.4 18.6-76.4 31.6 13 49.8 2.6 90-31.2 124-33.8 33.8-74.2 53.4-123.8 40.4-13.2 24-23.8 49.6-31.8 76.4 44.4 26 74 63.8 74 111.6s-29.6 94.8-74 120.8c8 26.8 18.6 52.4 31.6 76.4 49.8-13 90-2.6 123.8 31.2 34 33.8 44.2 74.2 31.2 124 24.2 13 49.6 23.6 76.4 31.6 26-44.4 73-73.8 120.8-73.8s94.8 29.4 120.8 73.8c26.8-8 52.4-18.6 76.4-31.6-13-49.8-2.6-90 31.2-124 33.8-33.8 74.2-53.4 124-40.4 13-24.2 23.6-49.8 31.6-76.4-44.2-26-73.6-63.8-73.6-111.6zM512 704c-105.8 0-192-86-192-192s86-192 192-192 192 86 192 192-86 192-192 192z"></path></svg></a>');
    });
});
