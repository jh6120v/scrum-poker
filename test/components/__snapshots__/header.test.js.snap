// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test header component. test <Header /> snapshot. 1`] = `
<header
  class="sc-jKJlTe qbGsk"
  data-testid="display_header"
>
  <div
    class="sc-hMqMXs fFHOfr"
    data-testid="display_prev_link"
  />
  <div
    class="sc-eNQAEJ emEcHI"
    data-testid="display_title"
  />
  <div
    class="sc-hMqMXs sc-kEYyzF ekJvCj"
    data-testid="display_next_link"
  />
</header>
`;
